mergify_engine/prom_exporter.py

```diff
@@ -80,7 +81,16 @@ def collect_metrics():
                 repositories, key=operator.attrgetter("private")):
             labels["private"] = private
             c = REPOSITORY_PER_INSTALLATION.labels(**labels)
-            c.set(len(list(repos)))
+            value = len(list(repos))
+            todo.append((c.set, (value, )))
+
+    # Prometheus can scrape data during our loop. So make it fast
+    # to ensure we always have the good values
+    INSTALLATIONS._metrics = []
+    REPOSITORY_PER_INSTALLATION._metrics = []
+    USERS_PER_INSTALLATION._metrics = []
+    for method, args in todo:
+        method(*args)
```

Comment:
This seems quite complicated. Why don't you just do:

```python
new_metrics = {}

<fill metric as before>

INSTALLATIONS._metrics = new_metrics
```

Reply:
_metrics is a list of so complicated object.